src/client/src/rt-util/hooks/useLocalStorage.ts

```diff
@@ -0,0 +1,31 @@
+import { useState, useEffect, useCallback } from 'react'
+
+export const useLocalStorage = <T>(key: string, initialState: T) => {
+    const [state, setState] = useState(() => {
+        return window.localStorage.getItem(key) ?? JSON.stringify(initialState)
+    })
+
+    useEffect(() => {
+        const updateState = () => {
+            setState(window.localStorage.getItem(key) ?? JSON.stringify(initialState))
+        }
+
+        window.addEventListener('storage', updateState)
+
+        return () => {
+            window.removeEventListener('storage', updateState)
+        }
+
+    }, [key, initialState])
+
+    useEffect(() => {
+        window.localStorage.setItem(key, state)
+    }, [key, state])
+
+    const value = JSON.parse(state)
+    const setter = useCallback((value: T) => {
+        setState(JSON.stringify(value))
```

Comment:
It would be nice to also be able to use the function update. Although, for now it's not necessary.

Reply:
Makes sense, updated.